mmcv/runner/runner.py

```diff
@@ -336,6 +337,15 @@ def run(self, data_loaders, workflow, max_epochs, **kwargs):
         self.logger.info('Start running, host: %s, work_dir: %s',
                          get_host_info(), work_dir)
         self.logger.info('workflow: %s, max: %d epochs', workflow, max_epochs)
+
+        # set epoch length, in case we use warmup_byepoch
+        # only one tuple in can be with mode 'train'
+        for i, flow in enumerate(workflow):
+            mode, epochs = flow
+            if mode == 'train':
+                self.epoch_len = len(data_loaders[i])
+                break
```

Comment:
No need to set `epoch_len` here, we can set it in the hook to reduce the influence to the runner.